msbuild/Xamarin.MacDev.Tasks/Tasks/UnpackLibraryResourcesTaskBase.cs

```diff
@@ -221,5 +239,45 @@ public Stream Open ()
 				return callback ();
 			}
 		}
+
+		public void Cancel ()
+		{
+			if (ShouldExecuteRemotely ())
+				BuildConnection.CancelAsync (BuildEngine4).Wait ();
+		}
+
+		public bool ShouldCopyToBuildServer (ITaskItem item)
+		{
+			if (item.IsFrameworkItem ())
+				return false;
+
+			if (NoOverwrite is not null && NoOverwrite.Contains (item))
+				return false;
+
+			return true;
+		}
+
+		public bool ShouldCreateOutputFile (ITaskItem item) => UnpackedResources.Contains (item) == true;
+
+		public IEnumerable<ITaskItem> GetAdditionalItemsToBeCopied () => Enumerable.Empty<ITaskItem> ();
+
+		IEnumerable<ManifestResource> GetAssemblyManifestResources (string fileName)
+		{
+			AssemblyDefinition? assembly = null;
+			try {
+				try {
+					assembly = AssemblyDefinition.ReadAssembly (fileName);
+				} catch {
+					yield break;
+				}
```

Comment:
Do you really need to have the nested try? can you just move the catch to the outer try and do the yield break there?

Reply:
Yes, this code can be simplified, but I wanted to keep the diff simple since it's just moving code to make it easier to review.